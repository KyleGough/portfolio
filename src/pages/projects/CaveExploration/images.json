[
  {
    "src": "/img/cave-exploration1.jpg",
    "caption": "Multiple drone cave exploration"
  },
  {
    "src": "/img/cave-exploration2.jpg",
    "caption": "Single drone cave exploration"
  },
  {
    "src": "/img/cave-exploration3.jpg",
    "caption": "Single drone cave exploration"
  },
  { "src": "/img/cave-generation1.png", "caption": "Cave generation step 1" },
  { "src": "/img/cave-generation2.png", "caption": "Cave generation step 2" },
  { "src": "/img/cave-generation3.png", "caption": "Cave generation step 3" },
  { "src": "/img/cave-generation4.png", "caption": "Cave generation step 4" }
]
