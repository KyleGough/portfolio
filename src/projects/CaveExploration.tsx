import { useEffect } from 'react';
import Section from '../components/Section';
import Divider from '../components/Divider';
import Chip from '../components/Chip';
import Pagination from '../components/Pagination';
import Figure from '../components/Figure';
import Screenshots from '../components/Screenshots';

export default function CaveExploration() {

    useEffect(() => {
        document.title = 'Cave Exploration - Improving and Simulating Cave Explortation with Swarm AI';
    }, []);

    const images = [
        { src:'/img/cave-exploration1.jpg', caption:'Multiple drone cave exploration' },
        { src:'/img/cave-exploration2.jpg', caption:'Single drone cave exploration' },
        { src:'/img/cave-exploration3.jpg', caption:'Single drone cave exploration' },
        { src:'/img/cave-generation1.png', caption:'Cave generation step 1' },
        { src:'/img/cave-generation2.png', caption:'Cave generation step 2' },
        { src:'/img/cave-generation3.png', caption:'Cave generation step 3' },
        { src:'/img/cave-generation4.png', caption:'Cave generation step 4' }
    ];

    return (
        <>
        <Section>
            <h1 className='project-title'>Using Swarm AI to Map a Cave Network</h1>
            <h2 className='project-subtitle'>CS310: Dissertation</h2>
            <p className='text-link-hover my-4'>December 2018 - April 2019</p>
            <p className='mb-4'>Cave exploration is dangerous and time-consuming. This project demonstrates how swarm AI could be used in a group of autonomous flying drones to navigate and explore a cave as efficiently as possible. The project also includes random, realistic cave environment generation.</p>
            <p className='mb-4'>Explore this project on 
                <a className='text-link hover:text-link-hover' href='https://github.com/KyleGough/CS310-Dissertation' target='_blank' rel='noopener noreferrer'> GitHub</a>
            </p>

            <div className='flex flex-row flex-wrap items-center mt-8'>
                <Chip name='C++' />
                <Chip name='OpenGL' />
            </div>
        </Section>

        <Divider />

        <Figure image={images[0]} />

        <Divider />

        <Section>
            <h2 className='project-header'>Features</h2>
            <ul className='project-list'>
                <li><strong>Realistic cave environment generation</strong> - Starting from Simplex noise, a cellular automata and several flood fills create unique, realistic cave environments for the simulation to run on.</li>
                <li><strong>Individual drone searching</strong> - A single drone can successfully explore every cell in the cave using a semi-efficient approach.</li>
                <li><strong>Multiple drone searching</strong> - Multiple drones can work together to more efficiently explore the cave by communicating information between them and avoiding exploring the same locations.</li>
                <li><strong>Visualisation</strong> - View the process of the drone's exploration. Ability to see what cells the drone has explored, potential frontier cells and the next target cell.</li>
                <li><strong>Statistics</strong> - View statistics of each drone including the distance they have travelled and percentage of the cave they have explored.</li>
            </ul>
        </Section>

        <Divider />

        <Screenshots images={images} />

        <Divider />
        
        <Pagination
            previousTitle='Roller Coaster'
            previousLink='/projects/roller-coaster'
            nextTitle='QuRVe'
            nextLink='/projects/qurve'
        />
        </>
    );
}