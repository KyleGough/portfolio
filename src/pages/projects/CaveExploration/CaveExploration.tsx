import React, { useEffect } from 'react';

import { Divider } from '../../../components/Divider';
import { ImageFigure } from '../../../components/ImageFigure';
import { Pagination } from '../../../components/Pagination';
import { ProjectHeader } from '../../../components/ProjectHeader';
import { Screenshots } from '../../../components/Screenshots';
import { Section } from '../../../components/Section';
import { getProjectData } from '../../../data';
import images from './images.json';

const project = getProjectData('cave-exploration');

export const CaveExploration: React.FC = () => {
  useEffect(() => {
    document.title =
      'Cave Exploration - Improving and Simulating Cave Explortation with Swarm AI';
  }, []);

  return (
    <>
      <ProjectHeader project={project} />

      <Divider />

      <ImageFigure image={images[0]} />

      <Divider />

      <Section>
        <h2 className="project-header">Features</h2>
        <ul className="project-list max-w-reading">
          <li className="mb-4">
            <strong>Realistic cave environment generation</strong>
            <br />
            Starting from Simplex noise, a cellular automata and several flood
            fills create unique, realistic cave environments for the simulation
            to run on.
          </li>
          <li className="mb-4">
            <strong>Individual drone searching</strong>
            <br />A single drone can successfully explore every cell in the cave
            using a semi-efficient approach.
          </li>
          <li className="mb-4">
            <strong>Multiple drone searching</strong>
            <br />
            Multiple drones can work together to more efficiently explore the
            cave by communicating information between them and avoiding
            exploring the same locations.
          </li>
          <li className="mb-4">
            <strong>Visualisation</strong>
            <br />
            View the process of the drone&apos;s exploration. Ability to see
            what cells the drone has explored, potential frontier cells and the
            next target cell.
          </li>
          <li className="mb-4">
            <strong>Statistics</strong>
            <br />
            View statistics of each drone including the distance they have
            travelled and percentage of the cave they have explored.
          </li>
        </ul>
      </Section>

      <Divider />

      <Screenshots images={images} />

      <Divider />

      <Pagination
        previousTitle="Roller Coaster"
        previousLink="/projects/roller-coaster"
        nextTitle="React Minesweeper"
        nextLink="/projects/react-minesweeper"
      />
    </>
  );
};
