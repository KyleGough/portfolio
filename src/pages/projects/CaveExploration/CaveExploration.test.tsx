import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { CaveExploration } from './CaveExploration';

describe('CaveExploration project page', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <CaveExploration />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
