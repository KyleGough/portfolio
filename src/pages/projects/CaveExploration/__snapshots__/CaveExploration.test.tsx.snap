// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaveExploration project page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      Using Swarm AI to Map a Cave Network
    </h1>
    <h2
      class="project-subtitle"
    >
      Dissertation
    </h2>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2018-12"
      >
        December 2018
      </time>
       
      -
       
      <time
        datetime="2019-04"
      >
        April 2019
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      Cave exploration is dangerous and time-consuming. This project demonstrates how swarm AI could be used in a group of autonomous flying drones to navigate and explore a cave as efficiently as possible. The project also includes random, realistic cave environment generation.
    </p>
    <p
      class="mb-4"
    >
      <span>
        Explore this project on 
      </span>
      <a
        class="text-link hover:text-link-hover focus:text-link-hover"
        href="https://github.com/KyleGough/CS310-Dissertation"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </p>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        C++
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        GLUT
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Multiple drone cave exploration"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/cave-exploration1.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Multiple drone cave exploration
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Features
    </h2>
    <ul
      class="project-list max-w-reading"
    >
      <li
        class="mb-4"
      >
        <strong>
          Realistic cave environment generation
        </strong>
        <br />
        Starting from Simplex noise, a cellular automata and several flood fills create unique, realistic cave environments for the simulation to run on.
      </li>
      <li
        class="mb-4"
      >
        <strong>
          Individual drone searching
        </strong>
        <br />
        A single drone can successfully explore every cell in the cave using a semi-efficient approach.
      </li>
      <li
        class="mb-4"
      >
        <strong>
          Multiple drone searching
        </strong>
        <br />
        Multiple drones can work together to more efficiently explore the cave by communicating information between them and avoiding exploring the same locations.
      </li>
      <li
        class="mb-4"
      >
        <strong>
          Visualisation
        </strong>
        <br />
        View the process of the drone's exploration. Ability to see what cells the drone has explored, potential frontier cells and the next target cell.
      </li>
      <li
        class="mb-4"
      >
        <strong>
          Statistics
        </strong>
        <br />
        View statistics of each drone including the distance they have travelled and percentage of the cave they have explored.
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Multiple drone cave exploration"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-exploration1.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Multiple drone cave exploration
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Single drone cave exploration"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-exploration2.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Single drone cave exploration
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Single drone cave exploration"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-exploration3.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Single drone cave exploration
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Cave generation step 1"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-generation1.png"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Cave generation step 1
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Cave generation step 2"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-generation2.png"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Cave generation step 2
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Cave generation step 3"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-generation3.png"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Cave generation step 3
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Cave generation step 4"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/cave-generation4.png"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Cave generation step 4
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/roller-coaster"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            Roller Coaster
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/react-minesweeper"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            React Minesweeper
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
